ksql-metastore/src/main/java/io/confluent/ksql/metastore/KsqlStream.java

```diff
@@ -47,6 +48,11 @@ public StructuredDataSource cloneWithTimeField(String timestampfieldName) {
     return new KsqlStream(dataSourceName, schema, keyField, newTimestampField.get(), ksqlTopic);
   }
 
+  @Override
+  public QueryId getPersistentQueryId() {
+    return new QueryId("CSAS_" + dataSourceName);
```

Comment:
For my understanding, what would be the `dataSourceName` in a statement like `CREATE STREAM enriched_pv as SELECT users.userid AS userid, pageid, region FROM pageviews LEFT JOIN users ON pageviews.userid = users.userid;`

Reply:
@apurvam it will be `enriched_pv`.
@dguy why not assign the query id where we create the query, in `PhysicalPlanBuilder`?

Reply:
because it wouldn't be the same across all engines. i.e., when we compact away the dropped commands the sequence is incorrect. This way the `QueryId` will be unique as we know we can't have 2 datasources with the same name. In the future when we introduce other persistent queries against the same datasource, we will need to also derive a unique key from them.